### PR TITLE
Add workflow to auto label new issues 

### PR DESCRIPTION
**Pull request by teezzan**
This PR helps add `triage` label to new issues raised.

### I created a workflow in `.github/workflows` and used [a simple issue labeller](https://github.com/marketplace/actions/simple-issue-labeler) to perform the labelling. It requires that `secrets.GITHUB_TOKEN` be present for it to work seemlessly.
---

## Related Issue
#10 

---

## Motivation and Context
Helps to keep the issues sorted. After which priority will be assigned to the issues.
---

## How Has This Been Manually Tested?
I am unable to test it as I can't create an issue from a fork. Help would be appreciated in this regard.

---

## Development Checklist:
**Types Of Changes:**

- [ ] Bug Fix(es)
  - [ ] A **_bug_** label has been added to the PR
- [ ] Contains breaking change(s)
  - 💡 This would force library users to change there code.  This would involve a public facing API change, behavior that would force a change to the users code base, or even an update in behavior that could force a change to the library users code base.  Sometimes this it is unclear if the change is a breaking change.  If it is unclear, reach out so we can discuss and investigate if it is indeed a breaking change.
- [x] Additional feature and/or behavior added.
  - [ ] An **_enhancement_** label has been added to the PR
- [x] I have changes related to workflows (CI/CD)
  - 💡 These kind of changes are only done by the project owner and maintainers that are allowed to make changes to workflows
  - [ ] An **_workflow_** label has been added to the PR

**Documentation:**
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
  - [ ] A **_documentation_** label has been added to the PR

**Testing:**
- [ ] My change requires unit tests to be written
  - [ ] I have added tests to cover my changes.
- [ ] I have manually tested my changes to the best of my ability

---

## PR Checklist
- [x] I have read the **CONTRIBUTING** document.
- [x] An issue exists and is linked to this PR.
- [ ] This PR is only for bringing changes from ***feature*** branches into the ***develop*** branch
- [x] My code follows the coding style of this project.
  - 💡 This is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with coding style that should be ignored or changed, reach out and lets discuss it.
- [ ] I have written unit tests to cover my changes.
- [ ] All unit tests passed locally.
  - 💡 Status checks are put in place to run unit tests every single time a change is pushed to a PR.  This does not mean that the tests pass in both the local and CI environment.

### CODE REVIEW

**Comment by teezzan** [comment]
Please check this out @CalvinWilkinson. I can't add a label and I can't test on my fork. What do you propose I do?

**Comment by CalvinWilkinson** [comment]
> Please check this out @CalvinWilkinson. I can't add a label and I can't test on my fork. What do you propose I do?


Currently the project has not reached **_v1.0.0_** yet and we are currently in preview.  So this means I have to manually fast forward the develop branch to keep it updated until I get to **_v1.0.0_**.  This means that the develop is behind.  I just fast forwarded the branch.  So just pull in the latest changes of the develop branch into your branch to get the latest if you could.

When it comes to workflows, usually I would just take care of that stuff but since you wanted to help and the workflow is so simple and does not depend on branch workflow, I figured it would ok to get some help.  :)

As for testing, when it comes to workflows, there is a cycle you can take to allow for testing of workflows.

1. Create a personal test repo
2. Create a branch to hold the workflow changes
3. Set the default branch of the repo settings to the same branch as the branch that contains the workflow changes
    - This will allow GitHub to pick up the workflow so it can be ran 
4. Run the workflow in the manner that is required to satisfy your workflow events
5. Repeat this to make sure your workflow is working correctly

Just create sample issues in your repo and your workflow "should" detect this and then simply add the triage label.  

If your up to it, it would be a great opportunity to learn how this all works.  You can create a test repo if you would like and setup the workflow there.  Follow the steps above and when you are confident that it is working, let me know and set the PR as ready for review. 🙂

Refer to the github documentation for [Setting the default branch](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch).

Once I am ready to approve the PR, and I merge it into the develop branch.  I will also manually migrate these changes into the latest preview branch.  

Let me know if you need anything else.

**Comment by teezzan** [comment]
I just merged upstream from `develop` and I tested the workflow. It works well.
Thanks for your help.

File: .github/workflows/auto-label-issue.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Auto Label issues
```
Comment (CalvinWilkinson) [inline .github/workflows/auto-label-issue.yml:1]: Please rename this to "Triage Issue"

File: .github/workflows/auto-label-issue.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  issues:
+    types:
+      - reopened
+      - opened
+jobs:
+  label_issues:
+    runs-on: ubuntu-latest
```
Comment (CalvinWilkinson) [inline .github/workflows/auto-label-issue.yml:9]: Have this run on `windows-latest`

```suggestion
    runs-on: windows-latest
```

File: .github/workflows/auto-label-issue.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    permissions:
+      issues: write
+    steps:
+      - name: Label issues
+        uses: andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414
+        with:
+          add-labels: "triage"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+
```
Comment (CalvinWilkinson) [inline .github/workflows/auto-label-issue.yml:19]: Remove this very last empty line.

End result should be just a single empty line at the very end of the file.

File: .github/workflows/auto-label-issue.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      - reopened
+      - opened
+jobs:
+  label_issues:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: Label issues
+        uses: andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414
```
Comment (CalvinWilkinson) [inline .github/workflows/auto-label-issue.yml:14]: Please change the version to reference the latest release tag instead of a commit hash.

```suggestion
        uses: andymckay/labeler@1.0.3
```

**Review by CalvinWilkinson — CHANGES_REQUESTED** [review]
If you would not mind, can you please rename the file to `triage-issue`

**Comment by teezzan** [comment]
All has been fixed. Thank you for the great learning experience.

**Review by CalvinWilkinson — APPROVED** [review]
Thanks for the help @teezzan!!
